reana_client/cli/files.py

```diff
@@ -534,6 +534,58 @@ def move_files(ctx, source, target, workflow, access_token):  # noqa: D301
         sys.exit(1)
 
 
+@files_group.command("prune")
+@check_connection
+@add_workflow_option
+@add_access_token_options
+@click.option(
+    "--preserve-reports",
+    "preserve_reports",
+    is_flag=True,
+    help="Don't delete the workflow-engine-produced output files "
+    "(this includes the Yadage-produced workflow animated gif, or the"
+    "Snakemake-produced report).",
+)
+@click.option(
+    "--include-all-runs",
+    "all_runs",
+    is_flag=True,
+    help="Prune the workspaces of all the runs of the workflow.",
+)
+@click.pass_context
+def prune_files(ctx, workflow, access_token, preserve_reports, all_runs):
+    r"""
+    Prune the workspace.
+
+    The ``prune`` command deletes all the files in the workflow that are not present
+    in the input or output section of the workflow definition.
+
+    Examples:\n
+    \t $ reana-client prune -w myanalysis.42
+    \t $ reana-client prune -w myanalysis.42 --include-all-runs
+    \t $ reana-client prune -w myanalysis.42 --preserve-reports
```

Comment:
I think we should always preserve reports, if we are preserving workflow outputs.

If a user would like to prune everything except inputs, perhaps the option would be better called `--include-outputs`.  

We could also add another option called `--include-inputs` which would prune also inputs. This will allow to really wipe out everything from the workspace if somebody wants:

```console
$ reana-client prune -w myanalysis --include-inputs --include-outputs --include-all-runs
```